addons/sale_loyalty/tests/test_program_numbers.py

```diff
@@ -1958,3 +1958,55 @@ def test_apply_order_and_specific_discounts(self):
             order.amount_total,
             "Order total should be 0, as a specific discount should have been applied.",
         )
+
+    def test_100_percent_discount_rounding(self):
+        self.env['loyalty.program'].create({
+            'name': '100% Discount',
+            'program_type': 'coupons',
+            'trigger': 'with_code',
+            'applies_on': 'current',
+            'rule_ids': [(0, 0, {
+                'code': '100pc',
+                'mode': 'with_code',
+            })],
+            'reward_ids': [(0, 0, {
+                'reward_type': 'discount',
+                'discount': 100,
+                'discount_mode': 'percent',
+                'discount_applicability': 'order',
+            })],
+        })
+
+        tax_9_99pc = self.env['account.tax'].create({
```

Comment:
```suggestion
        tax_19_99pc = self.env['account.tax'].create({
```
Also check other occurrences